webpack/assets/javascripts/react_app/components/TimeseriesChart/Normalizer.js

```diff
@@ -0,0 +1,24 @@
+
+const Normalizer = (vector) => {
+  const c3Resp = {
+    x: 'time',
+    columns: [['time']],
+  };
+  if (vector.length === 0) {
+    return c3Resp;
+  }
+
+  const timeFrame = vector[0].data.map(ts => ts[0]);
+  const tsItems = vector.map(item => [item.label, ...item.data.map(ts => ts[1])]);
+  c3Resp.columns = [['time', ...timeFrame], ...tsItems];
+
+  const colorItems = vector.filter(item => Object.prototype.hasOwnProperty.call(item, 'color'));
+  if (colorItems.length === 0) {
+    return c3Resp;
+  }
+  const colorReducer = (acc, cur) => Object.assign(acc, { [cur.label]: cur.color });
```

Comment:
be aware that `acc` is going to be mutated, if you'd like just a copy, use Object.assign({}, `copied data`), or use object spread operator, which I tend to prefer because of its readability.